velox/docs/develop/types.rst

```diff
@@ -2,61 +2,75 @@
 BOOLEAN                 bool                            0.125 (i.e. 1 bit)
 TINYINT                 int8_t                          1
 SMALLINT                int16_t                         2
 INTEGER                 int32_t	                        4
 BIGINT                  int64_t                         8
+HUGEINT                 int128_t                       16
 DATE                    struct Date                     8
 REAL                    float                           4
 DOUBLE                  double                          8
-DECIMAL                 int64_t / int128_t              8 / 16
 TIMESTAMP               struct Timestamp               16
-INTERVAL DAY TO SECOND  int64_t                        8
 VARCHAR                 struct StringView              16
 VARBINARY               struct StringView              16
 ======================  ===========================    ==================
 
-DECIMAL type is a logical type and carries additional `precision`,
+Logical Types
+~~~~~~~~~~~~~
+Velox logical types are backed by a physical type.
+The table below shows the supported logical types, and
+their corresponding physical type.
+
+======================  ===========================
+Velox Logical Type      Physical Type
+======================  ===========================
+DECIMAL                 BIGINT / HUGEINT
+INTERVAL DAY TO SECOND  BIGINT
+======================  ===========================
+
+DECIMAL type carries additional `precision`,
 and `scale` information. `Precision` is the number of
 digits in a number. `Scale` is the number of digits to the right of the decimal
 point in a number. For example, the number `123.45` has a precision of `5` and a
-scale of `2`. Decimal types are backed by `int64_t` and `int128_t` C++ types which
-store the unscaled value.
-`int64_t` is used upto 18 precision, and has a range of
-[:math:`-10^{18} + 1, +10^{18} - 1`]. `int128_t` is used starting from 19 precision
+scale of `2`. Decimal types are backed by `BIGINT` and `HUGEINT` physical types,
+which store the unscaled value. For example, the unscaled value of decimal
+`123.45` is `12345`. `BIGINT` is used upto 18 precision, and has a range of
+[:math:`-10^{18} + 1, +10^{18} - 1`]. `HUGEINT` is used starting from 19 precision
 upto 38 precision, with a range of [:math:`-10^{38} + 1, +10^{38} - 1`].
-For example, the unscaled value of decimal `123.45` is `12345`.
+
 All the three values, precision, scale, unscaled value are required to represent a
 decimal value.
-Some systems use two `int64_t` values instead of a single `int128_t` value.
-Velox chose `int128_t` since most compilers now support this type and
-it simplifies the implementation.
+
+Complex Types
+~~~~~~~~~~~~~
+Velox supports the ARRAY, MAP, and ROW complex types.
 
 Presto Types
 ~~~~~~~~~~~~
-Velox also supports certain types specific to Presto.
+Velox also supports certain types specific to Presto that are backed by a
```

Comment:
Let's say 'Presto supports a number of Presto-specific logical types.'

It will be obvious that these are backed by physical types as all logical types are.